### ISSUE TITLE
Implement LogSoftmax, Softmax, ReduceMax

### ISSUE THREAD

**Issue by jsilter** [zama-ai/concrete-ml#680]
## Feature request

Request the implementation of the following ONNX operators:
* LogSoftmax
* Softmax
* ReduceMax

## Motivation

These operators are common in neural networks of many types; softmax is common in classification, reducemax is common in CNNs.

**Comment by andrei-stoian-zama** [comment]
Thanks for the suggestion. We don't currently have plans to implement these ops. You can find a softmax implementation in the [LLM use case](https://github.com/zama-ai/concrete-ml/blob/f469f88e2944ed4da4d12307a648e2d18d301563/use_case_examples/llm/qgpt2_class.py#L233). There is already a ReduceSum operator and a MaxPool operator, you could try to adapt them to obtain ReduceMax.

**Comment by jsilter** [comment]
Thanks for the pointers. Do you have additional guides on how one implements a custom op in this context? I really would need to convert my existing (torch) model, I can't re-write and retrain a quantized version. 

It does seem like the reduce_sum could be adapted easily (np.sum -> np.amax) although I don't know if that's FHE compliant. Also I notice that there's already an implementation for Softmax, although the [docstring says it's not FHE compliant](https://github.com/zama-ai/concrete-ml/blob/f469f88e2944ed4da4d12307a648e2d18d301563/src/concrete/ml/onnx/ops_impl.py#L346) it doesn't elaborate as to why.

**Comment by jfrery** [comment]
We have this documentation / guide on how to proceed to implement a new onnx node. Let us know if anything is unclear!

https://docs.zama.ai/concrete-ml/developers/support_new_onnx_node
